tests/unit/commands/generate-test.js

```diff
@@ -47,9 +49,20 @@ describe('generate command', function() {
 
   it('runs GenerateFromBlueprint but with null nodeModulesPath', function() {
     command.project.hasDependencies = function() { return false; };
+    let originalYarnLockPath = path.join(command.project.root, 'yarn.lock');
+    let dummyYarnLockPath = path.join(command.project.root, 'foo.bar');
+    fs.renameSync(originalYarnLockPath, dummyYarnLockPath);
 
     return expect(command.validateAndRun(['controller', 'foo'])).to.be.rejected.then(reason => {
       expect(reason.message).to.eql('node_modules appears empty, you may need to run `npm install`');
+    }).then(() => fs.renameSync(dummyYarnLockPath, originalYarnLockPath));
```

Comment:
this `renameSync` will not run if the assertion above fails. you could wrap it in a describe() block though:

```js
describe('without yarn.lock file', function() {
  let originalYarnLockPath, dummyYarnLockPath;

  beforeEach(function() {
    originalYarnLockPath = path.join(command.project.root, 'yarn.lock');
    dummyYarnLockPath = path.join(command.project.root, 'foo.bar');
    fs.renameSync(originalYarnLockPath, dummyYarnLockPath);
  });

  afterEach(function() {
    fs.renameSync(dummyYarnLockPath, originalYarnLockPath);
  });

  it('runs GenerateFromBlueprint but with null nodeModulesPath', function() {
    command.project.hasDependencies = function() { return false; };

    return expect(command.validateAndRun(['controller', 'foo'])).to.be.rejected.then(reason => {
      expect(reason.message).to.eql('node_modules appears empty, you may need to run `npm install`');
    });
  });
});
``